pkg/verification/verification_service.go

```diff
@@ -57,28 +58,65 @@ func NewVerificationService(cfg ServiceConfiguration, opts ...VerificationServic
 	return s
 }
 
-// SendVerification sends a verification message to the specified user.  If successful, it will modify the
-// specified UserSignup parameter, setting the verification code annotation label to the value of the
-// generated verification code
-func (s *ServiceImpl) SendVerification(ctx *gin.Context, signup *v1alpha1.UserSignup) error {
-	verificationCode, err := generateVerificationCode()
+// InitVerification sends a verification message to the specified user.  If successful,
+// the user will receive a verification SMS.
+func (s *ServiceImpl) InitVerification(ctx *gin.Context, signup *v1alpha1.UserSignup, countryCode, phoneNumber string) (*v1alpha1.UserSignup, error) {
+	now := time.Now()
+
+	// If 24 hours has passed since the verification timestamp, then reset the timestamp and verification attempts
+	ts, err := time.Parse(TimestampLayout, signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey])
+	if err != nil || (err == nil && now.After(ts.Add(24*time.Hour))) {
+		// Set a new timestamp
+		signup.Annotations[v1alpha1.UserSignupVerificationInitTimestampAnnotationKey] = now.Format(TimestampLayout)
+		signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey] = "0"
+	}
+
+	// get the annotation counter
+	annotationCounter := signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey]
+	var counter int
+	if annotationCounter == "" {
+		counter = 0
+	}
+
+	dailyLimit := s.config.GetVerificationDailyLimit()
+	counter, err = strconv.Atoi(annotationCounter)
 	if err != nil {
-		return err
+		// We shouldn't get an error here, but if we do, we should probably set verification attempts to daily limit
+		// so that we at least now have a valid value
+		signup.Annotations[v1alpha1.UserSignupVerificationCounterAnnotationKey] = strconv.Itoa(dailyLimit)
+		return signup, errors3.NewInternalError(err, fmt.Sprintf("error when retrieving counter annotation for UserSignup %s, set to daily limit", signup.GetName()))
```

Comment:
the `errors3 "github.com/codeready-toolchain/registration-service/pkg/errors"` alias should be removed. `errors` (ie, default pkg name) is enough here

Reply:
fixed as part of: https://github.com/codeready-toolchain/registration-service/pull/126/commits/60ca7377aabe427ed8f8e3d642380b1b8396454e